### DISCUSSION TITLE
Could SWRV manage Apollo Client's cache?

### DISCUSSION THREAD

**Discussion by smolinari** [Kong/swrv#137] [Q&A]
Hey. 

this is a really nice caching tool. I was wondering though, could it be possible to give SWRV the Apollo cache to "manage"? To me, this is a missing piece of Apollo Client's almost finished puzzle. The manual updating of the cache for "optimistic UI" seems like it should be automatable and SWRV is sort of that kind of automation, from what I understand. Just wondering.... 😃 

Or, should SWRV just replace the cache? Hmmm.... 🤔 

Scott

**Comment by darrenjennings** [comment]
From what I can tell, the apollo client's strength is its caching and compat with the apollo server. With swrv, it is data fetching agnostic so you can do things like:

```ts
import { request } from 'graphql-request'

const fetcher = query => request('/api/graphql', query)

export default {
  setup () {
    const { data, error } = useSWR(`{
      Movie(title: "Inception") {
        releaseDate
        actors {
          name
        }
      }
    }`,
    fetcher)
  
  return { data, error }
}
```

Which is nice if you want to fetch from an apollo server endpoint in one component, and a REST endpoint in the next. [vue-apollo](https://github.com/vuejs/vue-apollo) (vue's apollo client) is tailor made for gql so from what I can tell, you might find it useful if you're consuming an entirely apollo server. If someone with more graphql knowledge wants to weigh in, I'd love to hear the pros/cons and use cases.

**Reply by smolinari** [reply]
> is its caching and compat with the apollo server

The Apollo Client isn't built in any way to be especially compatible with Apollo Server. It follows the GraphQL spec for the most part, so it can be a GraphQL client for any standard GraphQL server. 

> it is data fetching agnostic

I'd like to understand this better. Like how to offer SWRV or wrap it around Apollo Client? Is it as simple as offering the client.query or mutation as the `fetcher`?

> Which is nice if you want to fetch from an apollo server endpoint in one component, and a REST endpoint in the next.

You can do this anyway, with any component at any time. It would be just a question of how useful that would be, I guess. 

Scott
